akka-actor-typed/src/main/scala/akka/actor/typed/SpawnProtocol.scala

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed
+
+import akka.actor.typed.scaladsl.Behaviors
+
+object SpawnProtocol {
+
+  object Spawn {
+    /**
+     * Special factory to make using Spawn with ask easier
+     */
+    def apply[T](behavior: Behavior[T], name: String, props: Props): ActorRef[ActorRef[T]] ⇒ Spawn[T] =
+      replyTo ⇒ new Spawn(behavior, name, props, replyTo)
+  }
+
+  /**
+   * Spawn a child actor with the given `behavior` and send back the `ActorRef` of that child to the given
+   * `replyTo` destination.
+   *
+   * If `name` is an empty string an anonymous actor (with automatically generated name) will be created.
+   */
+  final case class Spawn[T](behavior: Behavior[T], name: String, props: Props, replyTo: ActorRef[ActorRef[T]])
+    extends SpawnProtocol {
+
+  }
+
+  /**
+   * Behavior implementing the [[SpawnProtocol]].
+   */
+  val behavior: Behavior[SpawnProtocol] =
+    Behaviors.receive { (ctx, msg) ⇒
+      msg match {
+        case Spawn(bhvr, name, props, replyTo) ⇒
+          val ref =
+            if (name == null || name.equals(""))
+              ctx.spawnAnonymous(bhvr, props)
+            else
+              ctx.spawn(bhvr, name, props)
```

Comment:
Should we handle that the name is already taken somehow?

Reply:
good point, but what could it do? should we always give them unique names by appending a counter?

Reply:
Not quite sure if it is best to handle or fail somehow. Names aren't as important here as in untyped where you'd actually use the path here and there, so maybe making unique is fine.

Reply:
we could use the given name if it's not taken already and if it is taken we append a number from a "global" counter (keeping track of one counter per name feels overkill)

Reply:
Sounds good